### ISSUE TITLE
Memory leak in MPLS label parsing

### ISSUE THREAD

**Issue by junekiwi** [cmand/yarrp#13]
I noticed that yarrp crashed on a 2GB VM when running moderately large traces (~200M targets). I haven't done a proper investigation but at first glance this seems to be due to the MPLS label stack allocated in `ICMP4::ICMP4()` never being freed.

From https://github.com/cmand/yarrp/blob/master/icmp.cpp#L115-L120:
```c++
mpls_label_t *lse = (mpls_label_t *) calloc(1, sizeof(mpls_label_t) ); // Never freed
mpls_stack = lse;
for (int labels = 0; labels < MAX_MPLS_STACK_HEIGHT; labels++) {
    tmp = (uint32_t *) ptr;
    if (labels > 0) {
        mpls_label_t *nextlse = (mpls_label_t *) calloc(1, sizeof(mpls_label_t) ); // Never freed
```

As a quick workaround I've commented out the MPLS code path and this solves the issue.

**Comment by robertbeverly** [comment]
Thanks a lot for reporting this.  

I've committed a fix that appears to work in `ccd16b3`.  Happy for any updates if this address the memory leak issue you're running into.

**Comment by junekiwi** [comment]
Thanks a lot for the quick fix @robertbeverly!
